airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/SentryExceptionHelper.java

```diff
@@ -166,4 +169,43 @@ private static Optional<List<SentryException>> buildJavaSentryExceptions(final S
     return Optional.of(sentryExceptions);
   }
 
+  private static Optional<List<SentryException>> buildNormalizationDbtSentryExceptions(final String stacktrace) {
+    final List<SentryException> sentryExceptions = new ArrayList<>();
+
+    // focus on Database Errors (these are the common errors seen in Airbyte)
+    final String databaseErrorIdentifier = "Database Error in model";
+    // for other dbt error types, we'll default to non-parsed stacktrace grouping
+    if (!stacktrace.contains(databaseErrorIdentifier)) {
+      return Optional.empty();
+    }
+
+    final String[] separatedStackTrace = stacktrace.split("\n");
+
+    // first let's get our useful error line for grouping
+    String usefulError = "";
+    boolean nextLine = false;
+    for (String errorLine : separatedStackTrace) {
+      // previous line was "Database Error..." so this is our useful message line
+      if (nextLine) {
+        usefulError = errorLine;
+        break;
```

Comment:
From reading this I think we're only building one SentryException from the stack trace, though I see the dbt stacks could have multiple errors (e.g. error 1 of 2 counts) - is there any value in building a SentryException for each of those?

Reply:
I played around with this but decided to scrap it because:
- the dbt error logs are json formatted and do _sometimes_ have an identifier (.node_info.unique_identifier), however this isn't present in every error log line so we can't rely on that for grouping.
- we also can't rely on simple order (for example look at the stacktrace in [this test event](https://sentry.io/organizations/g-c/issues/3506843909/?project=6459076&query=&statsPeriod=14d)). We could get clever and parse structures that are ordered but this would then still only cover the errors that are structured in a specific way when unfortunately they're wildly variable (e.g. check out [all these with different structures](https://airbyte.metabaseapp.com/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJxdWVyeSIsInF1ZXJ5Ijp7InNvdXJjZS10YWJsZSI6MTAsImZpbHRlciI6WyJhbmQiLFsiPSIsWyJmaWVsZCIsMjQ0NzAsbnVsbF0sIm5vcm1hbGl6YXRpb24tc3lzdGVtX2Vycm9yIl0sWyJkb2VzLW5vdC1jb250YWluIixbImZpZWxkIiwyNDQ3MixudWxsXSwiRGF0YWJhc2UgRXJyb3IgaW4gbW9kZWwiLHsiY2FzZS1zZW5zaXRpdmUiOmZhbHNlfV1dfSwiZGF0YWJhc2UiOjJ9LCJkaXNwbGF5IjoidGFibGUiLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7InRhYmxlLmNvbHVtbl93aWR0aHMiOltudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCwxMzc4LG51bGwsbnVsbCxudWxsLG51bGwsMjAyLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCxudWxsLG51bGwsbnVsbCw0OTBdfX0=)).

side note: I will be pushing up some changes soon to cover a wider array of the error structures but for now going to just identify events based on the first error message for simplicity. Will create an issue for incremental improvement on this to handle multiple exceptions.